Core.ApplicationServices/Organizations/StsOrganizationSynchronizationService.cs

```diff
@@ -223,5 +263,36 @@ private Maybe<OperationError> Modify(Guid organizationUuid, Func<Organization, M
 
             return Maybe<OperationError>.None;
         }
+
+        private Result<ExternalOrganizationUnit, OperationError> ConnectToExternalOrganizationHierarchy(
+            Organization organization, ExternalOrganizationUnit importRoot, Maybe<int> levelsToInclude, bool subscribeToUpdates)
+        {
+            return organization.ConnectToExternalOrganizationHierarchy(OrganizationUnitOrigin.STS_Organisation, importRoot, levelsToInclude, subscribeToUpdates)
+                .Match
+                (
+                    error => error,
+                    () => Result<ExternalOrganizationUnit, OperationError>.Success(importRoot));
+        }
+
+        private Result<StsOrganizationChangeLog, OperationError> WithLogEntries(OrganizationTreeUpdateConsequences consequences)
+        {
+            var changeLog = new StsOrganizationChangeLog { Origin = StsOrganizationChangeLogOrigin.Background };
+
+            if (_activeUserIdContext.HasValue)
+            {
+                var userId = _activeUserIdContext.Value.ActiveUserId;
+                var user = _userRepository.GetById(userId);
+                if (user == null)
```

Comment:
why are you checking an injected fact... the active user context is the authenticated user... if you don't trust that then you would need this check in every method in kitos dealing with auth (all methods almost)

Trust the id

Reply:
I am adding afix in the PR for this